homeassistant/components/sensor/recorder.py

```diff
@@ -543,3 +544,56 @@ def list_statistic_ids(hass: HomeAssistant, statistic_type: str | None = None) -
         statistic_ids[entity_id] = statistics_unit
 
     return statistic_ids
+
+
+def validate_statistics(
```

Comment:
Let's also use this logic inside `compile_statistics`. All it should do is to create a persistent notification that issues have been detected. No longer needed to log as we're going to add a UI.

Reply:
Yeah, let's address this is a follow-up 👍